<?xml version="1.0"?>

<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="my_robot">

<xacro:macro name="f_link" params="prefix parent length radius link_rpy link_xyz">

		<link name="${prefix}">
        <visual>
            <origin rpy="${link_rpy}" xyz="${link_xyz}"/>
            <geometry>
                <cylinder length="${length}" radius="${radius}"/>
            </geometry>
        </visual>
        <inertial>
            <origin rpy="${link_rpy}" xyz="${link_xyz}"/>
            <mass value="1"/>
            <inertia ixx="0.06" ixy="0.0" ixz="0.0" iyy="0.06" iyz="0.0" izz="0.25"/>
        </inertial>
    </link>
    
    <joint name="${parent}_to_${prefix}" type="fixed">
			<parent link="${parent}"/>
			<child link="${prefix}"/>
			<dynamics damping="0.5" friction="0.0"/>
		</joint>
		
</xacro:macro>
</robot>
